Download aspera@0.1.0

edsu7 opened the pull request.

edsu7 committed:
[wfpm v0.8.0] added starting template for download-aspera@0.1.0

edsu7 committed:
added main files and test cases

edsu7 committed:
Update pkg.json

lindaxiang commented on download-aspera/Dockerfile:
```diff
@@ -0,0 +1,34 @@
+FROM ubuntu:18.04
+
+LABEL org.opencontainers.image.source https://github.com/edsu7/argo-data-submission
```
use `icgc-argo` instead of personal repo: `https://github.com/icgc-argo/argo-data-submission` ?

lindaxiang commented on download-aspera/main.py:
```diff
@@ -0,0 +1,103 @@
+def randomword(length):
+   return(''.join(random.choice(string.ascii_lowercase) for i in range(length)))
+
+def mkdir_p(path,file):
```
Where is this function referenced?

edsu7 replied:
Sorry which function?
`randomword` is referenced in [L46](https://github.com/icgc-argo/argo-data-submission/blob/030a033a3062469a45cd3bad86bc3c3afc38c1a7/download-aspera/main.py#L46)
`mkdir_p` yeah I'll remove that

edsu7 replied:
Removed `mkdir_p`

https://github.com/icgc-argo/argo-data-submission/blob/025b0ab638e3dd968fb8d4a9b60fa839378c1fb7/download-aspera/main.py#L88-L92

lindaxiang commented on download-aspera/tests/checker.nf:
```diff
@@ -0,0 +1,106 @@
+
+    md5sum ${output_file} | egrep -o '^[0-9a-f]{32}' > normalized_output
+    echo "29df96fc47f09023bf00a044585fc697" > normalized_expected
+    
```
This is neat implementation. 👍

lindaxiang commented on download-aspera/main.nf:
```diff
@@ -0,0 +1,94 @@
+
+
+// tool specific parmas go here, add / change as needed
+params.input_file=''
```
A minor comment: `input_file` is kind of a misleading param name.

edsu7 replied:
Renamed `input_file` to `target_file`:
https://github.com/icgc-argo/argo-data-submission/blob/025b0ab638e3dd968fb8d4a9b60fa839378c1fb7/download-aspera/main.nf#L47

lindaxiang reviewed (changes requested):
Test it locally and it went well.

edsu7 committed:
Edits to address Linda's review

- changed Dockerfile label from personal to ICGC-argo
- renamed `input_file` to `target_file` in main.nf
- removed regex removal in main.nf to keep ".c4gh" suffix
- removed function "mkdir_p" in main.py. Unused

lindaxiang commented on download-aspera/tests/checker.nf:
```diff
@@ -0,0 +1,106 @@
+
+workflow checker {
+  take:
+    input_file
```
change `input_file` to `target_file` for consistence.

edsu7 replied:
Thanks for catching this. Fixed
https://github.com/icgc-argo/argo-data-submission/blob/ee937642d9394e5ddd35c548397905eb6ed1038b/download-aspera/tests/checker.nf#L75-L77

lindaxiang commented on download-aspera/tests/checker.nf:
```diff
@@ -0,0 +1,106 @@
+
+workflow {
+  checker(
+    params.input_file,
```
same as the above

edsu7 replied:
Fixed
https://github.com/icgc-argo/argo-data-submission/blob/ee937642d9394e5ddd35c548397905eb6ed1038b/download-aspera/tests/checker.nf#L98-L100

lindaxiang commented on download-aspera/tests/test-job-aspera.json:
```diff
@@ -0,0 +1,7 @@
+{
+    "input_file" : "/aspera-test-dir-large/100MB", 
```
change `input_file` to `target_file` for consistence.

lindaxiang commented on download-aspera/README.md:
````diff
@@ -0,0 +1,54 @@
+#### Contents of Param-file
+```
+{
+    "input_file" : "/aspera-test-dir-large/100MB", ### File to be download 
````
change `input_file` to `target_file` for consistence.

edsu7 replied:
Fixed.
https://github.com/icgc-argo/argo-data-submission/blob/ee937642d9394e5ddd35c548397905eb6ed1038b/download-aspera/tests/test-job-aspera.json#L2

edsu7 committed:
adding `target_file` to README.md, checker.nf and test-job-aspera
